web/client/plugins/TOC/components/StyleBasedWMSJsonLegend.jsx

```diff
@@ -104,6 +106,10 @@ class StyleBasedWMSJsonLegend extends React.Component {
         }
         this.setState({ loading: true });
         getJsonWMSLegend(jsonLegendUrl).then(data => {
+            const dynamicLegendIsEmpty = data.length === 0 || data[0].rules.length === 0;
+            if ((this.props.layer.dynamicLegendIsEmpty ?? null) !== dynamicLegendIsEmpty) {
+                this.props.onUpdateNode({ dynamicLegendIsEmpty });
+            }
```

Comment:
In https://github.com/allyoucanmap/MapStore2/commit/a11b6736dd0cc5175977ec75a212d3c7d205d309 same as previous comment